cmd/charm/charmcmd/cmd.go

```diff
@@ -181,7 +181,9 @@ func addAuthFlag(f *gnuflag.FlagSet, s *string) {
 
 // addChannelFlag adds the -c (--channel) flags to the given flag set.
 func addChannelFlag(f *gnuflag.FlagSet, s *string) {
-	f.StringVar(s, "c", string(params.StableChannel), "the channel the charm or bundle is assigned to")
+	f.StringVar(s, "c",
+		fmt.Sprint(string(params.StableChannel), `|`, string(params.DevelopmentChannel)),
```

Comment:
This doesn't seem right to me the third parameter is the default value, not the range of values. You should put the range in the help text in the  4th field. Otherwise you risk the value of s being "stable|development" which I would have thought charmstore would have a bad time with.